src/aspire/source/simulation.py

```diff
@@ -414,10 +414,13 @@ def eval_clustering(self, vol_idx):
         assert (
             len(vol_idx) == self.n
         ), f"Need {self.n} vol indexes to evaluate clustering"
-        # Remember that `states` is 1-indexed while vol_idx is 0-indexed
-        correctly_classified = np.sum(self.states - 1 == vol_idx)
+        # Remember that `states` is 1-indexed while vol_idx is 0-indexed.
+        # In addition, vol_idx may be improperly labeled so we check for
+        # the closest to 1-to-1 mapping between vol_idx and states.
+        remapped_vol_idx = (vol_idx + np.arange(self.C)[:, None]) % self.C
```

Comment:
Are you sure this is all the possible mappings... I'm not 100% but I think you may be forgetting `(self.C-1)!` of them.

Reply:
I'm definitely forgetting `(self.C - 1)!` mappings!